## 1. Import some Libraries, Read Dataset and Exploratory Dataset using Pandas

In [36]:
"""
Meng-import beberapa library dari bahasa pemrograman python yang bertujuan
untuk eksplorasi, membersihkan/mentransformasikan data serta pembuatan grafik visualisasi untuk di analisis
"""
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
plt.style.use("fast")
from sklearn.preprocessing import LabelEncoder

In [37]:
# Melakukan proses pembacaan dataset dari sub-folder "Dataset" serta menampilkan 17 data teratas
data_patient_treatment = pd.read_csv("../Dataset/patient_treatment_classification_dataset.csv")

data_patient_treatment.head(17)

HAEMATOCRIT  HAEMOGLOBINS  ERYTHROCYTE  LEUCOCYTE  THROMBOCYTE   MCH  \
0          35.1          11.8         4.65        6.3          310  25.4   
1          43.5          14.8         5.39       12.7          334  27.5   
2          33.5          11.3         4.74       13.2          305  23.8   
3          39.1          13.7         4.98       10.5          366  27.5   
4          30.9           9.9         4.23       22.1          333  23.4   
5          34.3          11.6         4.53        6.6          185  25.6   
6          31.1           8.7         5.06       11.1          416  17.2   
7          40.3          13.3         4.73        8.1          257  28.1   
8          33.6          11.5         4.54       11.4          262  25.3   
9          35.4          11.4         4.80        2.6          183  23.8   
10         33.7          11.5         4.57       13.2          322  25.2   
11         54.0          16.6         7.61       10.0           88  21.8   
12         31.7          10.4         4.91        9.7          348  21.2   
13         35.3          11.9         4.40        5.8          205  27.0   
14         34.5           9.8         5.75       15.4          548  17.0   
15         34.0          10.3         5.27       16.2          572  19.5   
16         35.0          11.6         4.58        7.4          154  25.3   

    MCHC   MCV  AGE SEX SOURCE  
0   33.6  75.5    1   F    out  
1   34.0  80.7    1   F    out  
2   33.7  70.7    1   F    out  
3   35.0  78.5    1   F    out  
4   32.0  73.0    1   M    out  
5   33.8  75.7    1   M    out  
6   28.0  61.5    1   F    out  
7   33.0  85.2    1   F    out  
8   34.2  74.0    1   F    out  
9   32.2  73.8    1   F    out  
10  34.1  73.7    1   M    out  
11  30.7  71.0    1   F     in  
12  32.8  64.6    1   M     in  
13  33.7  80.2    1   M    out  
14  28.4  60.0    1   M    out  
15  30.3  64.5    1   M    out  
16  33.1  76.4    1   F    out

Berikut adalah informasi atas penjelasan lengkap mengenai 11 kolom data yang ada di dalam variabel DataFrame yang bernama "data_patient_treatment" diantaranya adalah sebagai berikut:

1. HAEMATOCRIT: Kolom yang berisikan data terkait hasil uji laboratorium mengenai hematokrit pasien. Kolom data ini menunjukkan bagian dari pemeriksaan darah untuk mendeteksi apakah pasien tersebut mengalami anemia/kekurangan sel darah merah atau tidak.
2. HEMOGLOBINS: Kolom yang berisikan data terkait hasil uji laboratorium mengenai hemoglobin atau sel darah merah milik masing-masing pasien.
3. ERYTHROCYTE: Kolom yang berisikan data terkait hasil uji laboratorium mengenai eritrosit atau sel darah putih milik masing-masing pasien.
4. LEUCOCYTE: Kolom yang berisikan data terkait hasil uji laboratorium mengenai leukosit milik masing-masing pasien.
5. THROMBOCYTE: Kolom yang berisikan data terkait hasil uji laboratorium mengenai trombosit pada masing-masing pasien.
6. MCH: Kolom yang berisikan data terkait hasil uji laboratorium terhadap masing-masing pasien mengenai MCH atau Mean Corpuscular Hemoglobin). MCH merupakan jumlah rata-rata hemoglobin di dalam sel darah merah milik pasien.
7. MCHC: Kolom yang berisikan data terkait hasil uji lab pasien mengenai MCHC atau Mean Corpuscular Hemoglobin Concentration. MCHC merupakan perhitungan rata-rata terhadap konsentrasi hemoglobin di dalam eritrosit pasien.
8. MCV: Kolom yang berisikan data terkait hasil uji lab pasien mengenai MCV atau Mean Corpuscular Volume. MCV merupakan ukuran rata-rata dari sel darah merah milik pasien.
9. AGE: Kolom yang berisikan data mengenai umur dari setiap pasien.
10. SEX: Kolom yang berisikan data mengenai jenis kelamin pasien yang ditandai dengan nilai F (female/perempuan) dan M (male/laki-laki).
11. SOURCE: Terakhir adalah kolom yang berisikan data dari masing-masing status pasien apakah pasien tersebut merupakan pasien rawat inap (ditandai dengan data "in") atau pasien rawat jalan (ditulis/ditandai dengan data "out"). Kolom ini juga merupakan kolom target yang akan dijadikan sebagai label klasifikasi pada tahap pemodelan machine learning.

Selain informasi atas 11 kolom yang ada di dalam "data_patient_treatment" yang sudah dijelaskan di atas, kalian dapat mengunjungi sumber dataset dari Kaggle melalui link di bawah ini:

https://www.kaggle.com/datasets/saurabhshahane/patient-treatment-classification

Eksplorasi selanjutnya setelah menampilkan 17 data teratas dari "data_patient_treatment" adalah membuktikan apakah variabel "data_patient_treatment" merupakan sebuah DataFrame atau Series dengan cara mengetahui dimensi dari "data_patient_treatment". Jika dimensinya adalah 1 maka variabel "data_patient_treatment" merepresentasikan sebagai Series sedangkan jika dimensinya adalah 2 maka merepresentasikan sebagai DataFrame.

In [38]:
"""
Menampilkan nilai dimensi yang dimiliki oleh variabel "data_patient_treatment"
untuk mengetahui apakah variabel tersebut merupakan Series/DataFrame
"""
print("Nilai Dimensi dari Variabel 'data_patient_treatment' adalah (1: Series | 2: DataFrame): ", data_patient_treatment.ndim)

Nilai Dimensi dari Variabel 'data_patient_treatment' adalah (1: Series | 2: DataFrame):  2


In [39]:
# Menampilkan 5 data terbawah yang ada di dalam variabel DataFrame "data_patient_treatment"
data_patient_treatment.tail()

HAEMATOCRIT  HAEMOGLOBINS  ERYTHROCYTE  LEUCOCYTE  THROMBOCYTE   MCH  \
4407         32.8          10.4         3.49        8.1           72  29.8   
4408         33.7          10.8         3.67        6.7           70  29.4   
4409         33.2          11.2         3.47        7.2          235  32.3   
4410         31.5          10.4         3.15        9.1          187  33.0   
4411         33.5          10.9         3.44        5.8          275  31.7   

      MCHC    MCV  AGE SEX SOURCE  
4407  31.7   94.0   92   F     in  
4408  32.0   91.8   92   F     in  
4409  33.7   95.7   93   F    out  
4410  33.0  100.0   98   F     in  
4411  32.5   97.4   99   F    out

In [40]:
# Menampilkan jumlah baris dan kolom yang dimiliki oleh variabel "data_patient_treatment" secara terpisah
print("1. Menampilkan Jumlah Baris dan Kolom Data secara Terpisah: ")
print("Jumlah Baris Data yang Dimiliki oleh Variabel DataFrame 'data_patient_treatment': ", len(data_patient_treatment))

print("Jumlah Kolom yang Dimiliki oleh Variabel DataFrame 'data_patient_treatment': ",
      len(data_patient_treatment.columns), "\n")

# Menampilkan jumlah baris dan kolom dari variabel "data_patient_treatment" tanpa terpisah
print("2. Menampilkan Jumlah Baris dan Kolom tanpa Terpisah: ")
print("Jumlah Baris dan Kolom Data dari Variabel DataFrame 'data_patient_treatment': ", data_patient_treatment.shape)

1. Menampilkan Jumlah Baris dan Kolom Data secara Terpisah: 
Jumlah Baris Data yang Dimiliki oleh Variabel DataFrame 'data_patient_treatment':  4412
Jumlah Kolom yang Dimiliki oleh Variabel DataFrame 'data_patient_treatment':  11 

2. Menampilkan Jumlah Baris dan Kolom tanpa Terpisah: 
Jumlah Baris dan Kolom Data dari Variabel DataFrame 'data_patient_treatment':  (4412, 11)


In [41]:
# Menampilkan nama kolom dari variabel "data_patient_treatment"
data_patient_treatment.columns

Index(['HAEMATOCRIT', 'HAEMOGLOBINS', 'ERYTHROCYTE', 'LEUCOCYTE',
       'THROMBOCYTE', 'MCH', 'MCHC', 'MCV', 'AGE', 'SEX', 'SOURCE'],
      dtype='object')

Berdasarkan hasil diatas dapat diketahui jika variabel DataFrame "data_patient_treatment" memiliki 4.412 data yang tersebar ke dalam 11 kolom berbeda. Kesebelas kolom tersebut adalah HAEMATOCRIT, HAEMOGLOBINS, ERYTHROCYTE, LEUCOCYTE, THROMBOCYTE, MCH, MCHC, MCV, AGE, SEX dan SOURCE.

Namun jika diperhatikan penamaan kolom tersebut masih belum tepat serta ditulis menggunakan huruf kapital sehingga harus di-update menjadi format penamaan kolom yang sesuai. Proses transformasi terhadap penamaan kolom akan dilakukan pada bagian selanjutnya tepatnya pada bagian Data Preprocessing.

In [42]:
# Menampilkan tipe data apa saja yang dimiliki oleh variabel DataFrame "data_patient_treatment"
data_patient_treatment.dtypes

HAEMATOCRIT     float64
HAEMOGLOBINS    float64
ERYTHROCYTE     float64
LEUCOCYTE       float64
THROMBOCYTE       int64
MCH             float64
MCHC            float64
MCV             float64
AGE               int64
SEX              object
SOURCE           object
dtype: object

Berdasarkan hasil eksplorasi diatas menggunakan fungsi dtypes dari pandas, dapat diketahui bahwa setidaknya ada 3 tipe data yang berbeda yaitu float, int dan object. Tipe data float merupakan tipe data yang menunjukkan/identitas dari data berupa bilangan numerik pecahan. Tipe data int atau integer merupakan tipe data yang menjadi indetitas dari data numerik bilangan bulat. Sedangkan tipe data object merupakan tipe data yang menunjukkan identitas dari data kategorik atau data string seperti huruf, simbol karakter atau gabungan antara bilangan numerik (pecahan/bulat) dengan huruf/angka. 

In [43]:
# Menampilkan jumlah total dari elemen yang dimiliki oleh variabel DataFrame
data_patient_treatment.size

48532

Kenapa jumlah total element dari "data_patient_treatment" bernilai 48.532? Hal tersebut dikarenakan jumlah total element yang ada dimiliki oleh fungsi "size" merupakan sebuah bentuk bilangan hasil kalkulasi perkalian antara jumlah baris (number of rows) dengan jumlah kolom (number of columns) yang dimiliki oleh variabel DataFrame (dalam hal ini adalah data_patient_treatment).

Berarti jumlah baris dari "data_patient_treatment" yang bernilai 4.412 * jumlah kolom yang saat ini dimiliki oleh "data_patient_treatment" yaitu 11 sehingga menghasilkan 48.532 element. Untuk lebih jelasnya dapat lihat perhitungan yang ada di bawah ini:

4412 * 11 = 48532

In [44]:
# Menampilkan jumlah total data yang terindikasi sebagai data duplikat
data_patient_treatment.duplicated().sum()

0

Hasil diatas menunjukkan bahwa data yang dimiliki oleh "data_patient_treatment" tidak ada yang terduplikasi. Data duplikat merupakan data yang memiliki nilai yang sama di setiap kolomnya serta pada baris yang lain juga terdapat nilai data yang serupa. Selanjutnya saya akan mengecek apakah di dalam "data_patient_treatment" terdapat missing value di masing-masing kolom atau tidak dengan menggunakan fungsi "isnull()" dan "sum()".

In [45]:
"""
Menampilkan jumlah keseluruhan missing value yang dimiliki oleh
masing-masing kolom dari variabel DataFrame "data_patient_treatment"
"""
data_patient_treatment.isnull().sum()

HAEMATOCRIT     0
HAEMOGLOBINS    0
ERYTHROCYTE     0
LEUCOCYTE       0
THROMBOCYTE     0
MCH             0
MCHC            0
MCV             0
AGE             0
SEX             0
SOURCE          0
dtype: int64

In [52]:
"""
Menampilkan ringkasan informasi singkat yang dimiliki oleh DataFrame, baik informasi per masing-masing
kolom seperti missing value, tipe data, dll, informasi jumlah kolom & baris dan penggunaan memori
"""
data_patient_treatment.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4412 entries, 0 to 4411
Data columns (total 11 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   HAEMATOCRIT   4412 non-null   float64
 1   HAEMOGLOBINS  4412 non-null   float64
 2   ERYTHROCYTE   4412 non-null   float64
 3   LEUCOCYTE     4412 non-null   float64
 4   THROMBOCYTE   4412 non-null   int64  
 5   MCH           4412 non-null   float64
 6   MCHC          4412 non-null   float64
 7   MCV           4412 non-null   float64
 8   AGE           4412 non-null   int64  
 9   SEX           4412 non-null   object 
 10  SOURCE        4412 non-null   object 
dtypes: float64(7), int64(2), object(2)
memory usage: 379.3+ KB


In [54]:
# Menampilkan nilai statistik deskriptif untuk kolom berjenis numerik (float & int) yang ada di dalam "data_patient_treatment"
data_patient_treatment.describe()

HAEMATOCRIT  HAEMOGLOBINS  ERYTHROCYTE    LEUCOCYTE  THROMBOCYTE  \
count  4412.000000   4412.000000  4412.000000  4412.000000  4412.000000   
mean     38.197688     12.741727     4.541260     8.718608   257.524479   
std       5.974784      2.079903     0.784091     5.049041   113.972365   
min      13.700000      3.800000     1.480000     1.100000     8.000000   
25%      34.375000     11.400000     4.040000     5.675000   188.000000   
50%      38.600000     12.900000     4.570000     7.600000   256.000000   
75%      42.500000     14.200000     5.050000    10.300000   321.000000   
max      69.000000     18.900000     7.860000    76.600000  1183.000000   

               MCH         MCHC          MCV          AGE  
count  4412.000000  4412.000000  4412.000000  4412.000000  
mean     28.234701    33.343042    84.612942    46.626473  
std       2.672639     1.228664     6.859101    21.731218  
min      14.900000    26.000000    54.000000     1.000000  
25%      27.200000    32.700000    81.500000    29.000000  
50%      28.700000    33.400000    85.400000    47.000000  
75%      29.800000    34.100000    88.700000    64.000000  
max      40.800000    39.000000   115.600000    99.000000